Solr: can't use version 7.5.0 (only 7.2.1 works)

```
 Task :dbmodel:postProcess
Performing postprocess create-search-index.
[corePostProcess] SLF4J: Class path contains multiple SLF4J bindings.
[corePostProcess] SLF4J: Found binding in [jar:file:/home/shokin/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-simple/1.7.7/8095d0b9f7e0a9cd79a663c740e0f8fb31d0e2c8/slf4j-simple-1.7.7.jar!/org/slf4j/impl/StaticLoggerBinder.class]
[corePostProcess] SLF4J: Found binding in [jar:file:/home/shokin/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-jdk14/1.7.10/13675a57601d9106fe82314fe9eebb9b2d9b13c6/slf4j-jdk14-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
[corePostProcess] SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
[corePostProcess] SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
[corePostProcess] [Task worker for ':'] INFO com.zaxxer.hikari.HikariDataSource - db.production - Starting...
[corePostProcess] [Task worker for ':'] INFO com.zaxxer.hikari.HikariDataSource - db.production - Start completed.
[corePostProcess] Creating index for keyword search...
[corePostProcess] Creating keyword index failed
[corePostProcess] org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteExecutionException: Error from server at http://localhost:8983/solr/beanmine-search: error processing commands
[corePostProcess]       at org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteExecutionException.create(HttpSolrClient.java:829)
[corePostProcess]       at org.apache.solr.client.solrj.impl.HttpSolrClient.executeMethod(HttpSolrClient.java:620)
[corePostProcess]       at org.apache.solr.client.solrj.impl.HttpSolrClient.request(HttpSolrClient.java:255)
[corePostProcess]       at org.apache.solr.client.solrj.impl.HttpSolrClient.request(HttpSolrClient.java:244)
[corePostProcess]       at org.apache.solr.client.solrj.SolrRequest.process(SolrRequest.java:194)
[corePostProcess]       at org.apache.solr.client.solrj.SolrRequest.process(SolrRequest.java:211)
[corePostProcess]       at org.intermine.api.searchengine.solr.SolrIndexHandler.createFieldTypeDefinitions(SolrIndexHandler.java:304)
[corePostProcess]       at org.intermine.api.searchengine.solr.SolrIndexHandler.createIndex(SolrIndexHandler.java:65)
[corePostProcess]       at org.intermine.bio.postprocess.CreateSearchIndexProcess.postProcess(CreateSearchIndexProcess.java:66)
[corePostProcess]       at org.intermine.task.PostProcessorTask.execute(PostProcessorTask.java:70)
[corePostProcess]       at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
[corePostProcess]       at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[corePostProcess]       at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[corePostProcess]       at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[corePostProcess]       at java.lang.reflect.Method.invoke(Method.java:498)
[corePostProcess]       at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[corePostProcess]       at groovy.util.AntBuilder.performTask(AntBuilder.java:327)
[corePostProcess]       at groovy.util.AntBuilder.nodeCompleted(AntBuilder.java:272)
[corePostProcess]       at groovy.util.BuilderSupport.doInvokeMethod(BuilderSupport.java:154)
[corePostProcess]       at groovy.util.AntBuilder.doInvokeMethod(AntBuilder.java:207)
[corePostProcess]       at groovy.util.BuilderSupport.invokeMethod(BuilderSupport.java:67)
[corePostProcess]       at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:48)
[corePostProcess]       at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
[corePostProcess]       at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
[corePostProcess]       at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
[corePostProcess]       at org.intermine.plugin.postprocess.PostProcessPlugin$_apply_closure2$_closure6$_closure7.doCall(PostProcessPlugin.groovy:127)
[corePostProcess]       at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[corePostProcess]       at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[corePostProcess]       at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[corePostProcess]       at java.lang.reflect.Method.invoke(Method.java:498)
[corePostProcess]       at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
[corePostProcess]       at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
[corePostProcess]       at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
[corePostProcess]       at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
[corePostProcess]       at groovy.lang.Closure.call(Closure.java:414)
[corePostProcess]       at groovy.lang.Closure.call(Closure.java:430)
[corePostProcess]       at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2040)
[corePostProcess]       at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2025)
[corePostProcess]       at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2066)
[corePostProcess]       at org.codehaus.groovy.runtime.dgm$162.invoke(Unknown Source)
[corePostProcess]       at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:274)
[corePostProcess]       at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
[corePostProcess]       at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
[corePostProcess]       at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
[corePostProcess]       at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
[corePostProcess]       at org.intermine.plugin.postprocess.PostProcessPlugin$_apply_closure2$_closure6.doCall(PostProcessPlugin.groovy:67)
[corePostProcess]       at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[corePostProcess]       at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[corePostProcess]       at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[corePostProcess]       at java.lang.reflect.Method.invoke(Method.java:498)
[corePostProcess]       at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
[corePostProcess]       at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
[corePostProcess]       at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
[corePostProcess]       at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
[corePostProcess]       at groovy.lang.Closure.call(Closure.java:414)
[corePostProcess]       at groovy.lang.Closure.call(Closure.java:430)
[corePostProcess]       at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:681)
[corePostProcess]       at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:656)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
[corePostProcess]       at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
[corePostProcess]       at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
[corePostProcess]       at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
[corePostProcess]       at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:63)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
[corePostProcess]       at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
[corePostProcess]       at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
[corePostProcess]       at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
[corePostProcess]       at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
[corePostProcess]       at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
[corePostProcess]       at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
[corePostProcess]       at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
[corePostProcess]       at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
[corePostProcess]       at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
[corePostProcess]       at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
[corePostProcess]       at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
[corePostProcess]       at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
[corePostProcess]       at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
[corePostProcess]       at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
[corePostProcess]       at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
[corePostProcess]       at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
[corePostProcess]       at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
[corePostProcess]       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[corePostProcess]       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[corePostProcess]       at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
[corePostProcess]       at java.lang.Thread.run(Thread.java:748)


BUILD SUCCESSFUL in 7s
8 actionable tasks: 5 executed, 3 up-to-date
[Thread-41] INFO com.zaxxer.hikari.HikariDataSource - db.production - Shutdown initiated...
[Thread-41] INFO com.zaxxer.hikari.HikariDataSource - db.production - Shutdown completed.


AND

Performing postprocess create-autocomplete-index.
[corePostProcess] SLF4J: Class path contains multiple SLF4J bindings.
[corePostProcess] SLF4J: Found binding in [jar:file:/home/shokin/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-simple/1.7.7/8095d0b9f7e0a9cd79a663c740e0f8fb31d0e2c8/slf4j-simple-1.7.7.jar!/org/slf4j/impl/StaticLoggerBinder.class]
[corePostProcess] SLF4J: Found binding in [jar:file:/home/shokin/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-jdk14/1.7.10/13675a57601d9106fe82314fe9eebb9b2d9b13c6/slf4j-jdk14-1.7.10.jar!/org/slf4j/impl/StaticLoggerBinder.class]
[corePostProcess] SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
[corePostProcess] SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
[corePostProcess] [Task worker for ':'] INFO com.zaxxer.hikari.HikariDataSource - db.production - Starting...
[corePostProcess] [Task worker for ':'] INFO com.zaxxer.hikari.HikariDataSource - db.production - Start completed.
[corePostProcess] create autocomplete index ...
[corePostProcess] Indexing SOTerm.name
[corePostProcess] Indexing OntologyTerm.name


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dbmodel:postProcess'.
> org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteExecutionException: Error from server at http://localhost:8983/solr/beanmine-autocomplete: error processing commands

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':dbmodel:postProcess'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:63)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: : org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteExecutionException: Error from server at http://localhost:8983/solr/beanmine-autocomplete: error processing commands
        at org.intermine.task.PostProcessorTask.execute(PostProcessorTask.java:73)
        at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
        at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
        at org.intermine.plugin.postprocess.PostProcessPlugin$_apply_closure2$_closure6$_closure7.doCall(PostProcessPlugin.groovy:127)
        at org.intermine.plugin.postprocess.PostProcessPlugin$_apply_closure2$_closure6.doCall(PostProcessPlugin.groovy:67)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:681)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:656)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        ... 27 more
Caused by: org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteExecutionException: Error from server at http://localhost:8983/solr/beanmine-autocomplete: error processing commands
        at org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteExecutionException.create(HttpSolrClient.java:829)
        at org.apache.solr.client.solrj.impl.HttpSolrClient.executeMethod(HttpSolrClient.java:620)
        at org.apache.solr.client.solrj.impl.HttpSolrClient.request(HttpSolrClient.java:255)
        at org.apache.solr.client.solrj.impl.HttpSolrClient.request(HttpSolrClient.java:244)
        at org.apache.solr.client.solrj.SolrRequest.process(SolrRequest.java:194)
        at org.apache.solr.client.solrj.SolrRequest.process(SolrRequest.java:211)
        at org.intermine.web.autocompletion.AutoCompleter.buildIndex(AutoCompleter.java:244)
        at org.intermine.bio.postprocess.CreateAutocompleteIndexProcess.postProcess(CreateAutocompleteIndexProcess.java:53)
        at org.intermine.task.PostProcessorTask.execute(PostProcessorTask.java:70)
        ... 40 more


* Get more help at https://help.gradle.org

BUILD FAILED in 9s
8 actionable tasks: 5 executed, 3 up-to-date
[Thread-41] INFO com.zaxxer.hikari.HikariDataSource - db.production - Shutdown initiated...
[Thread-41] INFO com.zaxxer.hikari.HikariDataSource - db.production - Shutdown completed. 
```

--- a/intermine/api/src/main/java/org/intermine/api/searchengine/solr/SolrIndexHandler.java
+++ b/intermine/api/src/main/java/org/intermine/api/searchengine/solr/SolrIndexHandler.java
@@ -87,7 +87,7 @@ public final class SolrIndexHandler implements IndexHandler
             this.existingFieldTypes = getAllExisitingFieldTypesFromSolr(solrClient);
             this.existingCopyFields = getAllExisitingCopyFieldsFromSolr(solrClient);
 
-        } catch (SolrServerException e){
+        } catch (SolrServerException e) {
             LOG.error("Retrieving existing schema Definitions in Solr failed");
         }
 
@@ -213,9 +213,9 @@ public final class SolrIndexHandler implements IndexHandler
     private void addFieldNameToSchema(String fieldName, String fieldType, boolean stored,
                                      boolean indexed, SolrClient solrClient) throws IOException {
 
-        if (!this.indexedFields.contains(fieldName)){
-            if (this.existingFields != null){
-                if (!this.existingFields.contains(fieldName)){
+        if (!this.indexedFields.contains(fieldName)) {
+            if (this.existingFields != null) {
+                if (!this.existingFields.contains(fieldName)) {
                     Map<String, Object> fieldAttributes = new HashMap();
                     fieldAttributes.put("name", fieldName);
                     fieldAttributes.put("type", fieldType);
@@ -225,7 +225,8 @@ public final class SolrIndexHandler implements IndexHandler
                     fieldAttributes.put("required", false);
 
                     try {
-                        SchemaRequest.AddField schemaRequest = new SchemaRequest.AddField(fieldAttributes);
+                        SchemaRequest.AddField schemaRequest =
+                                new SchemaRequest.AddField(fieldAttributes);
                         SchemaResponse.UpdateResponse response = schemaRequest.process(solrClient);
 
                         this.indexedFields.add(fieldName);
@@ -244,8 +245,8 @@ public final class SolrIndexHandler implements IndexHandler
             throws IOException {
 
         try {
-            if (this.existingCopyFields != null){
-                if (!this.existingCopyFields.contains(dest)){
+            if (this.existingCopyFields != null) {
+                if (!this.existingCopyFields.contains(dest)) {
 
                     List<String> copyFieldAttributes = new ArrayList<String>();
                     copyFieldAttributes.add(dest);
@@ -293,8 +294,8 @@ public final class SolrIndexHandler implements IndexHandler
 
     private void createFieldTypeDefinitions(SolrClient solrClient) throws IOException {
 
-        if (this.existingFieldTypes != null){
-            if (!this.existingFieldTypes.contains(ANALYZED_FIELD_TYPE_NAME)){
+        if (this.existingFieldTypes != null) {
+            if (!this.existingFieldTypes.contains(ANALYZED_FIELD_TYPE_NAME)) {
                 FieldTypeDefinition analyzedFieldTypeDefinition = new FieldTypeDefinition();
 
                 Map<String, Object> analyzedFieldTypeAttributes = new HashMap();
@@ -309,7 +310,8 @@ public final class SolrIndexHandler implements IndexHandler
                 indexAnalyzerDefinition1.setTokenizer(indexTokenizerAttributes1);
                 Map<String, Object> indexLowerCaseFilterAttributes1 = new HashMap<String, Object>();
                 indexLowerCaseFilterAttributes1.put("class", "solr.LowerCaseFilterFactory");
-                List<Map<String, Object>> indexFilterAttributes1 = new ArrayList<Map<String, Object>>();
+                List<Map<String, Object>> indexFilterAttributes1 =
+                        new ArrayList<Map<String, Object>>();
                 indexFilterAttributes1.add(indexLowerCaseFilterAttributes1);
                 indexAnalyzerDefinition1.setFilters(indexFilterAttributes1);
 
@@ -319,7 +321,8 @@ public final class SolrIndexHandler implements IndexHandler
                 queryAnalyzerDefinition1.setTokenizer(queryTokenizerAttributes1);
                 Map<String, Object> queryLowerCaseFilterAttributes1 = new HashMap<String, Object>();
                 queryLowerCaseFilterAttributes1.put("class", "solr.LowerCaseFilterFactory");
-                List<Map<String, Object>> queryFilterAttributes1 = new ArrayList<Map<String, Object>>();
+                List<Map<String, Object>> queryFilterAttributes1 =
+                        new ArrayList<Map<String, Object>>();
                 queryFilterAttributes1.add(queryLowerCaseFilterAttributes1);
                 queryAnalyzerDefinition1.setFilters(queryFilterAttributes1);
 
@@ -342,8 +345,8 @@ public final class SolrIndexHandler implements IndexHandler
             }
         }
 
-        if (this.existingFieldTypes != null){
-            if (!this.existingFieldTypes.contains(RAW_FIELD_TYPE_NAME)){
+        if (this.existingFieldTypes != null) {
+            if (!this.existingFieldTypes.contains(RAW_FIELD_TYPE_NAME)) {
                 FieldTypeDefinition rawFieldTypeDefinition = new FieldTypeDefinition();
 
                 Map<String, Object> rawFieldTypeAttributes = new HashMap();
@@ -358,7 +361,8 @@ public final class SolrIndexHandler implements IndexHandler
                 indexAnalyzerDefinition2.setTokenizer(indexTokenizerAttributes2);
                 Map<String, Object> indexLowerCaseFilterAttributes2 = new HashMap<String, Object>();
                 indexLowerCaseFilterAttributes2.put("class", "solr.LowerCaseFilterFactory");
-                List<Map<String, Object>> indexFilterAttributes2 = new ArrayList<Map<String, Object>>();
+                List<Map<String, Object>> indexFilterAttributes2 =
+                        new ArrayList<Map<String, Object>>();
                 indexFilterAttributes2.add(indexLowerCaseFilterAttributes2);
                 indexAnalyzerDefinition2.setFilters(indexFilterAttributes2);
 
@@ -368,7 +372,8 @@ public final class SolrIndexHandler implements IndexHandler
                 queryAnalyzerDefinition2.setTokenizer(queryTokenizerAttributes2);
                 Map<String, Object> queryLowerCaseFilterAttributes2 = new HashMap<String, Object>();
                 queryLowerCaseFilterAttributes2.put("class", "solr.LowerCaseFilterFactory");
-                List<Map<String, Object>> queryFilterAttributes2 = new ArrayList<Map<String, Object>>();
+                List<Map<String, Object>> queryFilterAttributes2
+                        = new ArrayList<Map<String, Object>>();
                 queryFilterAttributes2.add(queryLowerCaseFilterAttributes2);
                 queryAnalyzerDefinition2.setFilters(queryFilterAttributes2);
 
@@ -398,8 +403,8 @@ public final class SolrIndexHandler implements IndexHandler
         List<String> allFields = new ArrayList<String>();
         SchemaRequest.Fields listFields = new SchemaRequest.Fields();
         SchemaResponse.FieldsResponse fieldsResponse = listFields.process(solrClient);
-        List<Map<String,Object>> solrFields = fieldsResponse.getFields();
-        for(Map<String,Object> field : solrFields) {
+        List<Map<String, Object>> solrFields = fieldsResponse.getFields();
+        for (Map<String, Object> field : solrFields) {
             allFields.add((String) field.get("name"));
         }
         return allFields;
@@ -411,7 +416,7 @@ public final class SolrIndexHandler implements IndexHandler
         SchemaRequest.FieldTypes listFieldTypes = new SchemaRequest.FieldTypes();
         SchemaResponse.FieldTypesResponse fieldTypesResponse = listFieldTypes.process(solrClient);
         List<FieldTypeRepresentation> solrFieldTypes = fieldTypesResponse.getFieldTypes();
-        for(FieldTypeRepresentation fieldType : solrFieldTypes) {
+        for (FieldTypeRepresentation fieldType : solrFieldTypes) {
             allFieldTypes.add((String) fieldType.getAttributes().get("name"));
         }
         return allFieldTypes;
@@ -423,7 +428,7 @@ public final class SolrIndexHandler implements IndexHandler
         SchemaRequest.CopyFields listCopyFields = new SchemaRequest.CopyFields();
         SchemaResponse.CopyFieldsResponse copyFieldsResponse = listCopyFields.process(solrClient);
         List<Map<String, Object>> solrCopyFields = copyFieldsResponse.getCopyFields();
-        for(Map<String, Object> copyField : solrCopyFields) {
+        for (Map<String, Object> copyField : solrCopyFields) {
             allCopyFields.add((String) copyField.get("dest"));
         }
         return allCopyFields;

--- a/intermine/api/src/main/java/org/intermine/api/searchengine/solr/SolrIndexHandler.java
+++ b/intermine/api/src/main/java/org/intermine/api/searchengine/solr/SolrIndexHandler.java
@@ -17,6 +17,7 @@ import org.apache.solr.client.solrj.request.schema.AnalyzerDefinition;
 import org.apache.solr.client.solrj.request.schema.FieldTypeDefinition;
 import org.apache.solr.client.solrj.request.schema.SchemaRequest;
 import org.apache.solr.client.solrj.response.UpdateResponse;
+import org.apache.solr.client.solrj.response.schema.FieldTypeRepresentation;
 import org.apache.solr.client.solrj.response.schema.SchemaResponse;
 import org.apache.solr.common.SolrInputDocument;
 import org.intermine.api.searchengine.IndexHandler;
@@ -49,6 +50,14 @@ public final class SolrIndexHandler implements IndexHandler
 
     private ObjectPipe<SolrInputDocument> indexingQueue = new ObjectPipe<SolrInputDocument>(100000);
 
+    //ArrayLists to store the existing schema data in Solr
+    //Based on the existence, schema will be created or updated
+    private List<String> existingFields;
+    private List<String> existingFieldTypes;
+    private List<String> existingCopyFields;
+
+    private List<String> indexedFields;
+
     @Override
     public void createIndex(ObjectStore os, Map<String, List<FieldDescriptor>> classKeys)
             throws IOException {
@@ -62,8 +71,6 @@ public final class SolrIndexHandler implements IndexHandler
         LOG.debug("Delete previous index begins");
         long deleteStartTime = System.currentTimeMillis();
 
-        createFieldTypeDefinitions(solrClient);
-
         try {
             solrClient.deleteByQuery("*:*");
             solrClient.commit();
@@ -74,6 +81,20 @@ public final class SolrIndexHandler implements IndexHandler
 
         LOG.debug("Delete previous index ends and it took "
                 + (System.currentTimeMillis() - deleteStartTime) + "ms");
+
+        try {
+            this.existingFields = getAllExistingFieldsFromSolr(solrClient);
+            this.existingFieldTypes = getAllExisitingFieldTypesFromSolr(solrClient);
+            this.existingCopyFields = getAllExisitingCopyFieldsFromSolr(solrClient);
+
+        } catch (SolrServerException e){
+            LOG.error("Retrieving existing schema Definitions in Solr failed");
+        }
+
+        this.indexedFields = new ArrayList<String>();
+
+        createFieldTypeDefinitions(solrClient);
+
 
         KeywordSearchPropertiesManager keywordSearchPropertiesManager
                 = KeywordSearchPropertiesManager.getInstance(os);
@@ -100,7 +121,9 @@ public final class SolrIndexHandler implements IndexHandler
                         keywordSearchPropertiesManager.getClassBoost(),
                         keywordSearchPropertiesManager.getFacets(),
                         keywordSearchPropertiesManager.getAttributePrefixes(),
-                        solrClient);
+                        solrClient,
+                        indexedFields,
+                        existingFields);
         fetchThread.start();
 
         int indexed = 0;
@@ -168,7 +191,6 @@ public final class SolrIndexHandler implements IndexHandler
 
     private void addSolrDocuments(SolrClient solrClient, List<SolrInputDocument> solrDocumentList)
             throws IOException {
-        //Accessing SchemaAPI from solr and create the schema dynamically
 
         if (solrDocumentList.size() != 0) {
 
@@ -191,45 +213,50 @@ public final class SolrIndexHandler implements IndexHandler
     private void addFieldNameToSchema(String fieldName, String fieldType, boolean stored,
                                      boolean indexed, SolrClient solrClient) throws IOException {
 
-        Map<String, Object> fieldAttributes = new HashMap();
-        fieldAttributes.put("name", fieldName);
-        fieldAttributes.put("type", fieldType);
-        fieldAttributes.put("stored", stored);
-        fieldAttributes.put("indexed", indexed);
-        fieldAttributes.put("multiValued", true);
-        fieldAttributes.put("required", false);
+        if (!this.indexedFields.contains(fieldName)){
+            if (this.existingFields != null){
+                if (!this.existingFields.contains(fieldName)){
+                    Map<String, Object> fieldAttributes = new HashMap();
+                    fieldAttributes.put("name", fieldName);
+                    fieldAttributes.put("type", fieldType);
+                    fieldAttributes.put("stored", stored);
+                    fieldAttributes.put("indexed", indexed);
+                    fieldAttributes.put("multiValued", true);
+                    fieldAttributes.put("required", false);
 
-        try {
-            SchemaRequest.AddField schemaRequest = new SchemaRequest.AddField(fieldAttributes);
-            SchemaResponse.UpdateResponse response = schemaRequest.process(solrClient);
+                    try {
+                        SchemaRequest.AddField schemaRequest = new SchemaRequest.AddField(fieldAttributes);
+                        SchemaResponse.UpdateResponse response = schemaRequest.process(solrClient);
 
-        } catch (SolrServerException e) {
-            LOG.error("Error while adding fields to the solrclient.", e);
+                        this.indexedFields.add(fieldName);
 
-            e.printStackTrace();
+                    } catch (SolrServerException e) {
+                        LOG.error("Error while adding fields to the solrclient.", e);
+
+                        e.printStackTrace();
+                    }
+                }
+            }
         }
-
     }
 
     private void addCopyFieldToSchema(String source, String dest, SolrClient solrClient)
             throws IOException {
 
         try {
+            if (this.existingCopyFields != null){
+                if (!this.existingCopyFields.contains(dest)){
 
-            List<String> copyFieldAttributes = new ArrayList<String>();
-            copyFieldAttributes.add(dest);
+                    List<String> copyFieldAttributes = new ArrayList<String>();
+                    copyFieldAttributes.add(dest);
 
-            SchemaRequest.DeleteCopyField deleteCopyField
-                    = new SchemaRequest.DeleteCopyField(source, copyFieldAttributes);
+                    SchemaRequest.AddCopyField schemaCopyRequest
+                            = new SchemaRequest.AddCopyField(source, copyFieldAttributes);
 
-            SchemaResponse.UpdateResponse deleteCopyFieldRes
-                    =  deleteCopyField.process(solrClient);
-
-            SchemaRequest.AddCopyField schemaCopyRequest
-                    = new SchemaRequest.AddCopyField(source, copyFieldAttributes);
-
-            SchemaResponse.UpdateResponse copyFieldResponse
-                    =  schemaCopyRequest.process(solrClient);
+                    SchemaResponse.UpdateResponse copyFieldResponse
+                            =  schemaCopyRequest.process(solrClient);
+                }
+            }
 
         } catch (SolrServerException e) {
             LOG.error("Error while adding copyfields to the solrclient.", e);
@@ -265,95 +292,141 @@ public final class SolrIndexHandler implements IndexHandler
     }
 
     private void createFieldTypeDefinitions(SolrClient solrClient) throws IOException {
-        FieldTypeDefinition analyzedFieldTypeDefinition = new FieldTypeDefinition();
 
-        Map<String, Object> analyzedFieldTypeAttributes = new HashMap();
-        analyzedFieldTypeAttributes.put("name", ANALYZED_FIELD_TYPE_NAME);
-        analyzedFieldTypeAttributes.put("class", "solr.TextField");
-        analyzedFieldTypeAttributes.put("positionIncrementGap", 100);
-        analyzedFieldTypeAttributes.put("multiValued", true);
+        if (this.existingFieldTypes != null){
+            if (!this.existingFieldTypes.contains(ANALYZED_FIELD_TYPE_NAME)){
+                FieldTypeDefinition analyzedFieldTypeDefinition = new FieldTypeDefinition();
 
-        AnalyzerDefinition indexAnalyzerDefinition1 = new AnalyzerDefinition();
-        Map<String, Object> indexTokenizerAttributes1 = new HashMap<String, Object>();
-        indexTokenizerAttributes1.put("class", "solr.WhitespaceTokenizerFactory");
-        indexAnalyzerDefinition1.setTokenizer(indexTokenizerAttributes1);
-        Map<String, Object> indexLowerCaseFilterAttributes1 = new HashMap<String, Object>();
-        indexLowerCaseFilterAttributes1.put("class", "solr.LowerCaseFilterFactory");
-        List<Map<String, Object>> indexFilterAttributes1 = new ArrayList<Map<String, Object>>();
-        indexFilterAttributes1.add(indexLowerCaseFilterAttributes1);
-        indexAnalyzerDefinition1.setFilters(indexFilterAttributes1);
+                Map<String, Object> analyzedFieldTypeAttributes = new HashMap();
+                analyzedFieldTypeAttributes.put("name", ANALYZED_FIELD_TYPE_NAME);
+                analyzedFieldTypeAttributes.put("class", "solr.TextField");
+                analyzedFieldTypeAttributes.put("positionIncrementGap", 100);
+                analyzedFieldTypeAttributes.put("multiValued", true);
 
-        AnalyzerDefinition queryAnalyzerDefinition1 = new AnalyzerDefinition();
-        Map<String, Object> queryTokenizerAttributes1 = new HashMap<String, Object>();
-        queryTokenizerAttributes1.put("class", "solr.WhitespaceTokenizerFactory");
-        queryAnalyzerDefinition1.setTokenizer(queryTokenizerAttributes1);
-        Map<String, Object> queryLowerCaseFilterAttributes1 = new HashMap<String, Object>();
-        queryLowerCaseFilterAttributes1.put("class", "solr.LowerCaseFilterFactory");
-        List<Map<String, Object>> queryFilterAttributes1 = new ArrayList<Map<String, Object>>();
-        queryFilterAttributes1.add(queryLowerCaseFilterAttributes1);
-        queryAnalyzerDefinition1.setFilters(queryFilterAttributes1);
+                AnalyzerDefinition indexAnalyzerDefinition1 = new AnalyzerDefinition();
+                Map<String, Object> indexTokenizerAttributes1 = new HashMap<String, Object>();
+                indexTokenizerAttributes1.put("class", "solr.WhitespaceTokenizerFactory");
+                indexAnalyzerDefinition1.setTokenizer(indexTokenizerAttributes1);
+                Map<String, Object> indexLowerCaseFilterAttributes1 = new HashMap<String, Object>();
+                indexLowerCaseFilterAttributes1.put("class", "solr.LowerCaseFilterFactory");
+                List<Map<String, Object>> indexFilterAttributes1 = new ArrayList<Map<String, Object>>();
+                indexFilterAttributes1.add(indexLowerCaseFilterAttributes1);
+                indexAnalyzerDefinition1.setFilters(indexFilterAttributes1);
 
-        analyzedFieldTypeDefinition.setAttributes(analyzedFieldTypeAttributes);
-        analyzedFieldTypeDefinition.setIndexAnalyzer(indexAnalyzerDefinition1);
-        analyzedFieldTypeDefinition.setQueryAnalyzer(queryAnalyzerDefinition1);
+                AnalyzerDefinition queryAnalyzerDefinition1 = new AnalyzerDefinition();
+                Map<String, Object> queryTokenizerAttributes1 = new HashMap<String, Object>();
+                queryTokenizerAttributes1.put("class", "solr.WhitespaceTokenizerFactory");
+                queryAnalyzerDefinition1.setTokenizer(queryTokenizerAttributes1);
+                Map<String, Object> queryLowerCaseFilterAttributes1 = new HashMap<String, Object>();
+                queryLowerCaseFilterAttributes1.put("class", "solr.LowerCaseFilterFactory");
+                List<Map<String, Object>> queryFilterAttributes1 = new ArrayList<Map<String, Object>>();
+                queryFilterAttributes1.add(queryLowerCaseFilterAttributes1);
+                queryAnalyzerDefinition1.setFilters(queryFilterAttributes1);
 
-        try {
-            SchemaRequest.AddFieldType schemaRequest
-                    = new SchemaRequest.AddFieldType(analyzedFieldTypeDefinition);
+                analyzedFieldTypeDefinition.setAttributes(analyzedFieldTypeAttributes);
+                analyzedFieldTypeDefinition.setIndexAnalyzer(indexAnalyzerDefinition1);
+                analyzedFieldTypeDefinition.setQueryAnalyzer(queryAnalyzerDefinition1);
 
-            SchemaResponse.UpdateResponse response =  schemaRequest.process(solrClient);
+                try {
+                    SchemaRequest.AddFieldType schemaRequest
+                            = new SchemaRequest.AddFieldType(analyzedFieldTypeDefinition);
 
-        } catch (SolrServerException e) {
-            LOG.error("Error while adding fieldtype '"
-                    + ANALYZED_FIELD_TYPE_NAME + "' to the solrclient.", e);
+                    SchemaResponse.UpdateResponse response =  schemaRequest.process(solrClient);
 
-            e.printStackTrace();
+                } catch (SolrServerException e) {
+                    LOG.error("Error while adding fieldtype '"
+                            + ANALYZED_FIELD_TYPE_NAME + "' to the solrclient.", e);
+
+                    e.printStackTrace();
+                }
+            }
         }
 
-        FieldTypeDefinition rawFieldTypeDefinition = new FieldTypeDefinition();
+        if (this.existingFieldTypes != null){
+            if (!this.existingFieldTypes.contains(RAW_FIELD_TYPE_NAME)){
+                FieldTypeDefinition rawFieldTypeDefinition = new FieldTypeDefinition();
 
-        Map<String, Object> rawFieldTypeAttributes = new HashMap();
-        rawFieldTypeAttributes.put("name", RAW_FIELD_TYPE_NAME);
-        rawFieldTypeAttributes.put("class", "solr.TextField");
-        rawFieldTypeAttributes.put("positionIncrementGap", 100);
-        rawFieldTypeAttributes.put("multiValued", true);
+                Map<String, Object> rawFieldTypeAttributes = new HashMap();
+                rawFieldTypeAttributes.put("name", RAW_FIELD_TYPE_NAME);
+                rawFieldTypeAttributes.put("class", "solr.TextField");
+                rawFieldTypeAttributes.put("positionIncrementGap", 100);
+                rawFieldTypeAttributes.put("multiValued", true);
 
-        AnalyzerDefinition indexAnalyzerDefinition2 = new AnalyzerDefinition();
-        Map<String, Object> indexTokenizerAttributes2 = new HashMap<String, Object>();
-        indexTokenizerAttributes2.put("class", "solr.KeywordTokenizerFactory");
-        indexAnalyzerDefinition2.setTokenizer(indexTokenizerAttributes2);
-        Map<String, Object> indexLowerCaseFilterAttributes2 = new HashMap<String, Object>();
-        indexLowerCaseFilterAttributes2.put("class", "solr.LowerCaseFilterFactory");
-        List<Map<String, Object>> indexFilterAttributes2 = new ArrayList<Map<String, Object>>();
-        indexFilterAttributes2.add(indexLowerCaseFilterAttributes2);
-        indexAnalyzerDefinition2.setFilters(indexFilterAttributes2);
+                AnalyzerDefinition indexAnalyzerDefinition2 = new AnalyzerDefinition();
+                Map<String, Object> indexTokenizerAttributes2 = new HashMap<String, Object>();
+                indexTokenizerAttributes2.put("class", "solr.KeywordTokenizerFactory");
+                indexAnalyzerDefinition2.setTokenizer(indexTokenizerAttributes2);
+                Map<String, Object> indexLowerCaseFilterAttributes2 = new HashMap<String, Object>();
+                indexLowerCaseFilterAttributes2.put("class", "solr.LowerCaseFilterFactory");
+                List<Map<String, Object>> indexFilterAttributes2 = new ArrayList<Map<String, Object>>();
+                indexFilterAttributes2.add(indexLowerCaseFilterAttributes2);
+                indexAnalyzerDefinition2.setFilters(indexFilterAttributes2);
 
-        AnalyzerDefinition queryAnalyzerDefinition2 = new AnalyzerDefinition();
-        Map<String, Object> queryTokenizerAttributes2 = new HashMap<String, Object>();
-        queryTokenizerAttributes2.put("class", "solr.KeywordTokenizerFactory");
-        queryAnalyzerDefinition2.setTokenizer(queryTokenizerAttributes2);
-        Map<String, Object> queryLowerCaseFilterAttributes2 = new HashMap<String, Object>();
-        queryLowerCaseFilterAttributes2.put("class", "solr.LowerCaseFilterFactory");
-        List<Map<String, Object>> queryFilterAttributes2 = new ArrayList<Map<String, Object>>();
-        queryFilterAttributes2.add(queryLowerCaseFilterAttributes2);
-        queryAnalyzerDefinition2.setFilters(queryFilterAttributes2);
+                AnalyzerDefinition queryAnalyzerDefinition2 = new AnalyzerDefinition();
+                Map<String, Object> queryTokenizerAttributes2 = new HashMap<String, Object>();
+                queryTokenizerAttributes2.put("class", "solr.KeywordTokenizerFactory");
+                queryAnalyzerDefinition2.setTokenizer(queryTokenizerAttributes2);
+                Map<String, Object> queryLowerCaseFilterAttributes2 = new HashMap<String, Object>();
+                queryLowerCaseFilterAttributes2.put("class", "solr.LowerCaseFilterFactory");
+                List<Map<String, Object>> queryFilterAttributes2 = new ArrayList<Map<String, Object>>();
+                queryFilterAttributes2.add(queryLowerCaseFilterAttributes2);
+                queryAnalyzerDefinition2.setFilters(queryFilterAttributes2);
 
-        rawFieldTypeDefinition.setAttributes(rawFieldTypeAttributes);
-        rawFieldTypeDefinition.setIndexAnalyzer(indexAnalyzerDefinition2);
-        rawFieldTypeDefinition.setQueryAnalyzer(queryAnalyzerDefinition2);
+                rawFieldTypeDefinition.setAttributes(rawFieldTypeAttributes);
+                rawFieldTypeDefinition.setIndexAnalyzer(indexAnalyzerDefinition2);
+                rawFieldTypeDefinition.setQueryAnalyzer(queryAnalyzerDefinition2);
 
-        try {
-            SchemaRequest.AddFieldType schemaRequest
-                    = new SchemaRequest.AddFieldType(rawFieldTypeDefinition);
+                try {
+                    SchemaRequest.AddFieldType schemaRequest
+                            = new SchemaRequest.AddFieldType(rawFieldTypeDefinition);
 
-            SchemaResponse.UpdateResponse response =  schemaRequest.process(solrClient);
+                    SchemaResponse.UpdateResponse response =  schemaRequest.process(solrClient);
 
-        } catch (SolrServerException e) {
-            LOG.error("Error while adding fieldtype '"
-                    +  RAW_FIELD_TYPE_NAME + "' to the solrclient.", e);
+                } catch (SolrServerException e) {
+                    LOG.error("Error while adding fieldtype '"
+                            +  RAW_FIELD_TYPE_NAME + "' to the solrclient.", e);
 
-            e.printStackTrace();
+                    e.printStackTrace();
+                }
+            }
         }
+    }
+
+
+    private List<String> getAllExistingFieldsFromSolr(SolrClient solrClient)
+            throws IOException, SolrServerException {
+        List<String> allFields = new ArrayList<String>();
+        SchemaRequest.Fields listFields = new SchemaRequest.Fields();
+        SchemaResponse.FieldsResponse fieldsResponse = listFields.process(solrClient);
+        List<Map<String,Object>> solrFields = fieldsResponse.getFields();
+        for(Map<String,Object> field : solrFields) {
+            allFields.add((String) field.get("name"));
+        }
+        return allFields;
+    }
+
+    private List<String> getAllExisitingFieldTypesFromSolr(SolrClient solrClient)
+            throws IOException, SolrServerException {
+        List<String> allFieldTypes = new ArrayList<String>();
+        SchemaRequest.FieldTypes listFieldTypes = new SchemaRequest.FieldTypes();
+        SchemaResponse.FieldTypesResponse fieldTypesResponse = listFieldTypes.process(solrClient);
+        List<FieldTypeRepresentation> solrFieldTypes = fieldTypesResponse.getFieldTypes();
+        for(FieldTypeRepresentation fieldType : solrFieldTypes) {
+            allFieldTypes.add((String) fieldType.getAttributes().get("name"));
+        }
+        return allFieldTypes;
+    }
+
+    private List<String> getAllExisitingCopyFieldsFromSolr(SolrClient solrClient)
+            throws IOException, SolrServerException {
+        List<String> allCopyFields = new ArrayList<String>();
+        SchemaRequest.CopyFields listCopyFields = new SchemaRequest.CopyFields();
+        SchemaResponse.CopyFieldsResponse copyFieldsResponse = listCopyFields.process(solrClient);
+        List<Map<String, Object>> solrCopyFields = copyFieldsResponse.getCopyFields();
+        for(Map<String, Object> copyField : solrCopyFields) {
+            allCopyFields.add((String) copyField.get("dest"));
+        }
+        return allCopyFields;
     }
 
 }

--- a/intermine/api/src/main/java/org/intermine/api/searchengine/solr/SolrObjectHandler.java
+++ b/intermine/api/src/main/java/org/intermine/api/searchengine/solr/SolrObjectHandler.java
@@ -122,6 +122,8 @@ public class SolrObjectHandler extends Thread
      *            addition to the normal indexing
      * @param attributePrefixes prefixes to be ignored
      * @param solrClient solrClient Instance
+     * @param indexedFields List of fieldnames that are indexed in the current postprocess
+     * @param existingFields List of fieldnames that are already exisiting
      */
     SolrObjectHandler(ObjectStore os, Map<String, List<FieldDescriptor>> classKeys,
                       ObjectPipe<SolrInputDocument> indexingQueue,
@@ -687,9 +689,9 @@ public class SolrObjectHandler extends Thread
         if (!fieldNames.contains(fieldName)) {
             fieldNames.add(fieldName);
 
-            if (!indexedFields.contains(fieldName)){
+            if (!indexedFields.contains(fieldName)) {
                 if (existingFields != null) {
-                    if (!existingFields.contains(fieldName)){
+                    if (!existingFields.contains(fieldName)) {
                         Map<String, Object> fieldAttributes = new HashMap();
                         fieldAttributes.put("name", fieldName);
                         fieldAttributes.put("type", fieldType);
@@ -699,8 +701,10 @@ public class SolrObjectHandler extends Thread
                         fieldAttributes.put("required", false);
 
                         try {
-                            SchemaRequest.AddField schemaRequest = new SchemaRequest.AddField(fieldAttributes);
-                            SchemaResponse.UpdateResponse response =  schemaRequest.process(solrClient);
+                            SchemaRequest.AddField schemaRequest
+                                    = new SchemaRequest.AddField(fieldAttributes);
+                            SchemaResponse.UpdateResponse response
+                                    = schemaRequest.process(solrClient);
 
                             indexedFields.add(fieldName);
 

--- a/intermine/api/src/main/java/org/intermine/api/searchengine/solr/SolrObjectHandler.java
+++ b/intermine/api/src/main/java/org/intermine/api/searchengine/solr/SolrObjectHandler.java
@@ -93,6 +93,9 @@ public class SolrObjectHandler extends Thread
 
     private SolrClient solrClient;
 
+    private List<String> indexedFields;
+    private List<String> existingFields;
+
     private volatile Exception error;
 
     /**
@@ -127,7 +130,9 @@ public class SolrObjectHandler extends Thread
                       Map<Class<? extends InterMineObject>, String[]> specialReferences,
                       Map<ClassDescriptor, Float> classBoost, Vector<KeywordSearchFacetData> facets,
                       Map<String, String> attributePrefixes,
-                      SolrClient solrClient
+                      SolrClient solrClient,
+                      List<String> indexedFields,
+                      List<String> existingFields
     ) {
         super();
 
@@ -142,6 +147,8 @@ public class SolrObjectHandler extends Thread
         this.attributePrefixes = attributePrefixes;
 
         this.solrClient = solrClient;
+        this.indexedFields = indexedFields;
+        this.existingFields = existingFields;
     }
 
     /**
@@ -680,25 +687,32 @@ public class SolrObjectHandler extends Thread
         if (!fieldNames.contains(fieldName)) {
             fieldNames.add(fieldName);
 
-            Map<String, Object> fieldAttributes = new HashMap();
-            fieldAttributes.put("name", fieldName);
-            fieldAttributes.put("type", fieldType);
-            fieldAttributes.put("stored", stored);
-            fieldAttributes.put("indexed", indexed);
-            fieldAttributes.put("multiValued", true);
-            fieldAttributes.put("required", false);
+            if (!indexedFields.contains(fieldName)){
+                if (existingFields != null) {
+                    if (!existingFields.contains(fieldName)){
+                        Map<String, Object> fieldAttributes = new HashMap();
+                        fieldAttributes.put("name", fieldName);
+                        fieldAttributes.put("type", fieldType);
+                        fieldAttributes.put("stored", stored);
+                        fieldAttributes.put("indexed", indexed);
+                        fieldAttributes.put("multiValued", true);
+                        fieldAttributes.put("required", false);
 
-            try {
-                SchemaRequest.AddField schemaRequest = new SchemaRequest.AddField(fieldAttributes);
-                SchemaResponse.UpdateResponse response =  schemaRequest.process(solrClient);
+                        try {
+                            SchemaRequest.AddField schemaRequest = new SchemaRequest.AddField(fieldAttributes);
+                            SchemaResponse.UpdateResponse response =  schemaRequest.process(solrClient);
 
-            } catch (Exception e) {
-                LOG.error("Error while adding fields to the solrclient.", e);
+                            indexedFields.add(fieldName);
 
-                e.printStackTrace();
+                        } catch (Exception e) {
+                            LOG.error("Error while adding fields to the solrclient.", e);
+
+                            e.printStackTrace();
+                        }
+                    }
+                }
             }
         }
-
     }
 
 }

--- a/intermine/webapp/src/main/java/org/intermine/web/autocompletion/AutoCompleter.java
+++ b/intermine/webapp/src/main/java/org/intermine/web/autocompletion/AutoCompleter.java
@@ -235,14 +235,14 @@ public class AutoCompleter
         try {
             existingFields = getAllExistingFieldsFromSolr(solrClient);
 
-        } catch (SolrServerException e){
+        } catch (SolrServerException e) {
             LOG.error("Retrieving existing Fieldnames in Solr failed");
         }
 
 
         for (String fieldName: fieldList) {
-            if (existingFields != null){
-                if (!existingFields.contains(fieldName)){
+            if (existingFields != null) {
+                if (!existingFields.contains(fieldName)) {
                     Map<String, Object> fieldAttributes = new HashMap();
                     fieldAttributes.put("name", fieldName);
                     fieldAttributes.put("type", "text_general");
@@ -252,7 +252,8 @@ public class AutoCompleter
                     fieldAttributes.put("required", false);
 
                     try {
-                        SchemaRequest.AddField schemaRequest = new SchemaRequest.AddField(fieldAttributes);
+                        SchemaRequest.AddField schemaRequest
+                                = new SchemaRequest.AddField(fieldAttributes);
                         SchemaResponse.UpdateResponse response =  schemaRequest.process(solrClient);
 
                     } catch (SolrServerException e) {
@@ -316,8 +317,8 @@ public class AutoCompleter
         List<String> allFields = new ArrayList<String>();
         SchemaRequest.Fields listFields = new SchemaRequest.Fields();
         SchemaResponse.FieldsResponse fieldsResponse = listFields.process(solrClient);
-        List<Map<String,Object>> solrFields = fieldsResponse.getFields();
-        for(Map<String,Object> field : solrFields) {
+        List<Map<String, Object>> solrFields = fieldsResponse.getFields();
+        for (Map<String, Object> field : solrFields) {
             allFields.add((String) field.get("name"));
         }
         return allFields;

--- a/intermine/webapp/src/main/java/org/intermine/web/autocompletion/AutoCompleter.java
+++ b/intermine/webapp/src/main/java/org/intermine/web/autocompletion/AutoCompleter.java
@@ -230,22 +230,36 @@ public class AutoCompleter
             e.printStackTrace();
         }
 
+        List<String> existingFields = null;
+
+        try {
+            existingFields = getAllExistingFieldsFromSolr(solrClient);
+
+        } catch (SolrServerException e){
+            LOG.error("Retrieving existing Fieldnames in Solr failed");
+        }
+
+
         for (String fieldName: fieldList) {
-            Map<String, Object> fieldAttributes = new HashMap();
-            fieldAttributes.put("name", fieldName);
-            fieldAttributes.put("type", "text_general");
-            fieldAttributes.put("stored", true);
-            fieldAttributes.put("indexed", true);
-            fieldAttributes.put("multiValued", true);
-            fieldAttributes.put("required", false);
+            if (existingFields != null){
+                if (!existingFields.contains(fieldName)){
+                    Map<String, Object> fieldAttributes = new HashMap();
+                    fieldAttributes.put("name", fieldName);
+                    fieldAttributes.put("type", "text_general");
+                    fieldAttributes.put("stored", true);
+                    fieldAttributes.put("indexed", true);
+                    fieldAttributes.put("multiValued", true);
+                    fieldAttributes.put("required", false);
 
-            try {
-                SchemaRequest.AddField schemaRequest = new SchemaRequest.AddField(fieldAttributes);
-                SchemaResponse.UpdateResponse response =  schemaRequest.process(solrClient);
+                    try {
+                        SchemaRequest.AddField schemaRequest = new SchemaRequest.AddField(fieldAttributes);
+                        SchemaResponse.UpdateResponse response =  schemaRequest.process(solrClient);
 
-            } catch (SolrServerException e) {
-                LOG.error("Error while adding autocomplete fields to the solrclient.", e);
-                e.printStackTrace();
+                    } catch (SolrServerException e) {
+                        LOG.error("Error while adding autocomplete fields to the solrclient.", e);
+                        e.printStackTrace();
+                    }
+                }
             }
         }
 
@@ -297,5 +311,16 @@ public class AutoCompleter
         }
     }
 
+    private List<String> getAllExistingFieldsFromSolr(SolrClient solrClient)
+            throws IOException, SolrServerException {
+        List<String> allFields = new ArrayList<String>();
+        SchemaRequest.Fields listFields = new SchemaRequest.Fields();
+        SchemaResponse.FieldsResponse fieldsResponse = listFields.process(solrClient);
+        List<Map<String,Object>> solrFields = fieldsResponse.getFields();
+        for(Map<String,Object> field : solrFields) {
+            allFields.add((String) field.get("name"));
+        }
+        return allFields;
+    }
 
 }

